add support of RustyHermit's BSD socket layer

stlankes opened the pull request:
RustyHermit is a tier 3 platform and publishes a new kernel interface. The new version supports a common BSD socket layer. By supporting this interface, the implementation of `std` can be harmonized to other operating systems. In `sys_common/mod.rs` we remove only a special case for RustyHermit. All changes are done in the RustyHermit specific directories.

To realize this socket layer, the handling of file descriptors is also harmonized to other operating systems.

stlankes committed:
add support of RustyHermit's BSD socket layer

RustHermit publishs a new kernel interface and supports
a common BSD socket layer. By supporting this interface,
the implementation can be harmonized to other operating systems.

To realize this socket layer, the handling of file descriptors
is also harmonized to other operating systems.

bjorn3 commented on Cargo.lock:
```diff
@@ -1934,8 +1934,16 @@ version = "0.2.6"
+
+[[package]]
+name = "hermit-abi"
+version = "0.3.0"
```
We will now have all three versions of hermit-abi required at the same time.

stlankes replied:
Oh, you are right. I will write PRs to upgrade `num_cpus` etc.

stlankes committed:
move IO traits to std/src/os/hermit

By moving the IO traits, the RustyHermit support is harmonized to
of other operating systems.

stlankes commented:
The last commit is more a suggestion from my side. I am not sure, where I should store the AsFd and similar traits. I trie to harmonize my implementation to other implementations.

stlankes committed:
remove code duplications

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4037385081/jobs/6940644766) [(plain)](https://github.com/rust-lang/rust/commit/68cef7cd8cc25abb86347138ad7df153bc411da1/checks/10958295001/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:ac593985615ec2ede58e132d2e21d2b1cbd6127c)
Download action repository 'rust-lang/simpleinfra@master' (SHA:ece894d15649f0f9d7884f915dc821f00bd0418b)
Complete job name: PR (mingw-check-tidy, true, ubuntu-20.04-xl)
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: mingw-check-tidy
---
  Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
Collecting boolean-py==4.0
  Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
Collecting chardet==5.1.0
  Downloading chardet-5.1.0-py3-none-any.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.1/199.1 KB 9.3 MB/s eta 0:00:00
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 KB 242.8 MB/s eta 0:00:00
  Downloading license_expression-30.0.0-py3-none-any.whl (86 kB)
  Downloading license_expression-30.0.0-py3-none-any.whl (86 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 86.4/86.4 KB 254.6 MB/s eta 0:00:00
  Downloading MarkupSafe-2.1.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting python-debian==0.1.49
  Downloading python_debian-0.1.49-py3-none-any.whl (132 kB)
  Downloading python_debian-0.1.49-py3-none-any.whl (132 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.5/132.5 KB 268.6 MB/s eta 0:00:00
  Downloading reuse-1.1.0.tar.gz (217 kB)
  Downloading reuse-1.1.0.tar.gz (217 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 217.0/217.0 KB 287.7 MB/s eta 0:00:00
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
  Downloading setuptools-66.0.0-py3-none-any.whl (1.3 MB)
  Downloading setuptools-66.0.0-py3-none-any.whl (1.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 266.3 MB/s eta 0:00:00
Building wheels for collected packages: reuse
  Building wheel for reuse (pyproject.toml): started
  Building wheel for reuse (pyproject.toml): finished with status 'done'
  Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=180117 sha256=a2b5d39c8ff2686626c851b00c3d3ae10157feb2cc6d0d07e414234b479dbb17
  Stored in directory: /tmp/pip-ephem-wheel-cache-ussj66fv/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
Successfully built reuse
Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
  Attempting uninstall: setuptools
    Found existing installation: setuptools 59.6.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'setuptools'. No files were found to uninstall.
Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
 ---> 59cb6c9b4e1d
Step 8/10 : COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
 ---> 7bf500e096fc
Step 9/10 : COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
---
Successfully built 2dda61c56d47
Successfully tagged rust-ci:latest
Built container sha256:2dda61c56d4712f60a22071dc814dc80d677825553ba25df997f8a08820b9723
Uploading finished image to https://ci-caches.rust-lang.org/docker/1ffa4bcb45050a94703a18527901adf2ca39095bf2769231379237670da45b38d77a6896233357c7a905df4ee90a3eae78692fb8b0a8a8fd5d59ce264a16a010
upload failed: - to s3://rust-lang-ci-sccache2/docker/1ffa4bcb45050a94703a18527901adf2ca39095bf2769231379237670da45b38d77a6896233357c7a905df4ee90a3eae78692fb8b0a8a8fd5d59ce264a16a010 Unable to locate credentials
[CI_JOB_NAME=mingw-check-tidy]
[CI_JOB_NAME=mingw-check-tidy]
---
configure: rust.debug-assertions := True
configure: rust.overflow-checks := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
Attempting with retry: make prepare
---
Highest error code: `E0792`
* 395 features
fmt check
skip untracked path cpu-usage.csv during rustfmt invocations
Diff in /checkout/library/std/src/os/hermit/io/mod.rs at line 1:
 #![stable(feature = "os_fd", since = "1.66.0")]
+mod net;
+mod net;
 #[path = "../../fd/owned.rs"]
 mod owned;
 #[path = "../../fd/raw.rs"]
Diff in /checkout/library/std/src/os/hermit/io/mod.rs at line 6:
 mod raw;
-mod net;
 // Export the types and traits for the public API.
 // Export the types and traits for the public API.
 #[stable(feature = "os_fd", since = "1.66.0")]
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/hermit/io/net.rs" "/checkout/library/std/src/os/wasi/net/mod.rs" "/checkout/library/std/src/os/unix/raw.rs" "/checkout/library/std/src/os/unix/process.rs" "/checkout/library/std/src/os/unix/mod.rs" "/checkout/library/std/src/os/wasi/mod.rs" "/checkout/library/std/src/os/unix/ucred/tests.rs" "/checkout/library/std/src/os/hermit/io/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
```

</details>

Mark-Simulacrum commented:
Please mark with `@rustbot ready` when this is ready for review -- looks like maybe some dependency upgrades are still in progress?

stlankes commented:
Ok, I will do it. However `is-terminal` depends on `io-lifetimes`. To update `io-lifetimes` I need the `fd` traits from the current PR. (https://github.com/hermitcore/io-lifetimes/commit/dd05273e4c13f9d8a757489f8ec865fc972cb703).

stlankes commented:
@Mark-Simulacrum If I understand it correctly, `atty` is not longer maintained and not all crates moved to `is-terminal`. In addition, `is-terminal` depends indirectly from this PR (see above). How do you handle it?

Mark-Simulacrum commented:
@rustbot ready

Mark-Simulacrum commented:
@bjorn3 is there a reason I'm missing we shouldn't just land this PR, and then follow-up with dropping hermit-abi dependencies once the ecosystem syncs back up?

bjorn3 commented:
That would work.

The changes in files not specific to hermit look correct to me. For the hermit specific files I didn't notice anything obviously wrong, but also didn't review it as closely. Given that RustyHermit is a tier 3 target and the PR author is a maintainer of RustyHermit, I think this is fine.

@bors r+

bors commented:
:pushpin: Commit dd270f1f440831cd4b491185b2fa2c35b6bfd503 has been approved by `bjorn3`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=bjorn3 dd270f1f440831cd4b491185b2fa2c35b6bfd503 -->
<!-- homu: {"type":"Approved","sha":"dd270f1f440831cd4b491185b2fa2c35b6bfd503","approver":"bjorn3","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit dd270f1f440831cd4b491185b2fa2c35b6bfd503 with merge 3103afcd69f7155552cd4c91f644f1c2c4a8628c...
<!-- homu: {"type":"BuildStarted","head_sha":"dd270f1f440831cd4b491185b2fa2c35b6bfd503","merge_sha":"3103afcd69f7155552cd4c91f644f1c2c4a8628c"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4158297208/jobs/7193486991)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4158297208/jobs/7193486991","builder_name":"checks-actions"} -->

bjorn3 commented:
@bors retry spurious failure writing an archive file during LLVM building

rust-log-analyzer commented:
The job **`dist-x86_64-mingw`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4158297208/jobs/7193417847) [(plain)](https://github.com/rust-lang-ci/rust/commit/3103afcd69f7155552cd4c91f644f1c2c4a8628c/checks/11285798839/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[2603/3025] Linking CXX static library lib\libLLVMMIRParser.a
[2604/3025] Building CXX object tools/lli/CMakeFiles/lli.dir/lli.cpp.obj
[2605/3025] Linking CXX static library lib\libLLVMInterpreter.a
FAILED: lib/libLLVMInterpreter.a 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E rm -f lib\libLLVMInterpreter.a && D:\a\rust\rust\mingw64\bin\ar.exe qc lib\libLLVMInterpreter.a  lib/ExecutionEngine/Interpreter/CMakeFiles/LLVMInterpreter.dir/Execution.cpp.obj lib/ExecutionEngine/Interpreter/CMakeFiles/LLVMInterpreter.dir/ExternalFunctions.cpp.obj lib/ExecutionEngine/Interpreter/CMakeFiles/LLVMInterpreter.dir/Interpreter.cpp.obj && D:\a\rust\rust\mingw64\bin\ranlib.exe lib\libLLVMInterpreter.a && cd ."
D:\a\rust\rust\mingw64\bin\ar.exe: could not create temporary file whilst writing archive: no more archived files
[2606/3025] Linking CXX static library lib\libLLVMCoroutines.a
[2607/3025] Linking CXX static library lib\libLLVMWebAssemblyUtils.a
[2608/3025] Linking CXX static library lib\libLLVMAsmPrinter.a
[2609/3025] Building CXX object tools/dsymutil/CMakeFiles/dsymutil.dir/DebugMap.cpp.obj
[2609/3025] Building CXX object tools/dsymutil/CMakeFiles/dsymutil.dir/DebugMap.cpp.obj
[2610/3025] Building CXX object tools/dsymutil/CMakeFiles/dsymutil.dir/dsymutil.cpp.obj
[2611/3025] Linking CXX static library lib\libLLVMSelectionDAG.a
[2612/3025] Linking CXX executable bin\llvm-profdata.exe
ninja: build stopped: subcommand failed.
command did not execute successfully, got: exit code: 1


build script failed, must exit now', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cmake-0.1.48\src\lib.rs:975:5
 finished in 225.635 seconds
Build completed unsuccessfully in 0:06:21
```

</details>

stlankes commented:
Do you understand the `homu` error?

matthiaskrgr commented:
@bors rollup=iffy bumping up in the queue a bit

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4211153127/jobs/7309493238)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4211153127/jobs/7309493238","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`dist-i686-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4211153127/jobs/7309321156) [(plain)](https://github.com/rust-lang-ci/rust/commit/ae3be874798ec322f669bf03cdc0d132066fe88d/checks/11436865237/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
warning:                                               ^
warning: 1 warning generated.
[RUSTC-TIMING] profiler_builtins test:false 0.038
 Documenting std v0.0.0 (D:\a\rust\rust\library\std)
error[E0432]: unresolved imports `libc::STDERR_FILENO`, `libc::STDIN_FILENO`, `libc::STDOUT_FILENO`
   |
   |
21 | use libc::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
   |            ^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^ no `STDOUT_FILENO` in the root
   |            |              no `STDIN_FILENO` in the root
   |            |              no `STDIN_FILENO` in the root
   |            no `STDERR_FILENO` in the root
error: Compilation failed, aborting rustdoc

For more information about this error, try `rustc --explain E0432`.
error: could not document `std`
error: could not document `std`

Caused by:
  process didn't exit successfully: `D:\a\rust\rust\build\bootstrap\debug\rustdoc --edition=2021 --crate-type dylib --crate-type rlib --crate-name std library\std\src\lib.rs --target i686-pc-windows-msvc -o D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\doc --cfg "feature=\"addr2line\"" --cfg "feature=\"backtrace\"" --cfg "feature=\"compiler-builtins-c\"" --cfg "feature=\"gimli-symbolize\"" --cfg "feature=\"miniz_oxide\"" --cfg "feature=\"object\"" --cfg "feature=\"panic_unwind\"" --cfg "feature=\"profiler\"" --cfg "feature=\"profiler_builtins\"" --cfg "feature=\"std_detect_dlsym_getauxval\"" --cfg "feature=\"std_detect_file_io\"" -Zunstable-options --check-cfg names() --check-cfg values() --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -Z unstable-options --resource-suffix 1.69.0 --markdown-css rust.css --markdown-no-toc --index-page D:\a\rust\rust\src/doc/index.md -C metadata=b26be8d05ae850e0 -L dependency=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps -L dependency=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\release\deps --extern addr2line=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libaddr2line-e9f077627cb9c69b.rmeta --extern alloc=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\liballoc-5af9f30c122a4822.rmeta --extern cfg_if=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libcfg_if-5d2923dc127a6dde.rmeta --extern compiler_builtins=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libcompiler_builtins-22158058821fcd35.rmeta --extern core=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libcore-dd96ddbcbdf8007d.rmeta --extern hashbrown=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libhashbrown-69bb38b1ee85451f.rmeta --extern libc=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\liblibc-dbb88388557f8b8e.rmeta --extern miniz_oxide=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libminiz_oxide-d8216e455a6ef819.rmeta --extern object=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libobject-8c902dd1d3855193.rmeta --extern panic_abort=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libpanic_abort-ec71198c27530aee.rmeta --extern panic_unwind=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libpanic_unwind-00b0629b74f36222.rmeta --extern profiler_builtins=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libprofiler_builtins-126947f869acca2b.rmeta --extern rustc_demangle=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\librustc_demangle-015e6a0f523186b6.rmeta --extern std_detect=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libstd_detect-501ebbc7758fa336.rmeta --extern unwind=D:\a\rust\rust\build\i686-pc-windows-msvc\stage2-std\i686-pc-windows-msvc\doc\i686-pc-windows-msvc\release\deps\libunwind-366c628b709a4919.rmeta -Csymbol-mangling-version=legacy -Zunstable-options -Zunstable-options --check-cfg=values(bootstrap) --check-cfg=values(stdarch_intel_sde) --check-cfg=values(no_fp_fmt_parse) --check-cfg=values(no_global_oom_handling) --check-cfg=values(no_rc) --check-cfg=values(no_sync) --check-cfg=values(freebsd12) --check-cfg=values(backtrace_in_libstd) "--check-cfg=values(target_env,\"libnx\")" "--check-cfg=values(target_arch,\"asmjs\",\"spirv\",\"nvptx\",\"xtensa\")" -Dwarnings -Wrustdoc::invalid_codeblock_attributes --crate-version "1.69.0-nightly
  (ae3be8747
  2023-02-18)" "-Zcrate-attr=doc(html_root_url=\"https://doc.rust-lang.org/nightly/\")" --cfg backtrace_in_libstd` (exit code: 1)
```

</details>

mkroening commented on library/std/src/os/fd/raw.rs:
```diff
@@ -12,11 +15,20 @@ use crate::os::unix::io::OwnedFd;
 #[cfg(target_os = "wasi")]
 use crate::os::wasi::io::OwnedFd;
 use crate::sys_common::{AsInner, IntoInner};
+#[cfg(target_os = "hermit")]
+use hermit_abi::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+#[cfg(not(target_os = "hermit"))]
+use libc::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
```
The CI issue happens when documenting std on Windows. `std::os::fd` exists even on Windows with `cfg(doc)`. We might work around like this:

```suggestion
#[cfg(target_os = "hermit")]
use hermit_abi as libc;
```

This, together with changing the `FILENO`s back to `libc::*_FILENO`, should solve the windows docs issue.

stlankes committed:
avoid the usage of libc during the creation of documentation

stlankes commented:
@bjorn3 I think that @mkroening's comment is correct. The documentation was broken.

bjorn3 commented:
@bors r+

bors commented:
:pushpin: Commit af8ee641a84d9b56c97ecc1deddcd2c7aa5fc4d4 has been approved by `bjorn3`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=bjorn3 af8ee641a84d9b56c97ecc1deddcd2c7aa5fc4d4 -->
<!-- homu: {"type":"Approved","sha":"af8ee641a84d9b56c97ecc1deddcd2c7aa5fc4d4","approver":"bjorn3","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit af8ee641a84d9b56c97ecc1deddcd2c7aa5fc4d4 with merge 34e6673a0473e90ef01a18eb575392c9e3859747...
<!-- homu: {"type":"BuildStarted","head_sha":"af8ee641a84d9b56c97ecc1deddcd2c7aa5fc4d4","merge_sha":"34e6673a0473e90ef01a18eb575392c9e3859747"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4271551104/jobs/7437292311)
Approved by: bjorn3
Pushing 34e6673a0473e90ef01a18eb575392c9e3859747 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"bjorn3","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4271551104/jobs/7437292311"},"merge_sha":"34e6673a0473e90ef01a18eb575392c9e3859747"} -->

rust-timer commented:
Finished benchmarking commit (34e6673a0473e90ef01a18eb575392c9e3859747): [comparison URL](https://perf.rust-lang.org/compare.html?start=31448badfd74ea72d2c8622cc60d3dca889ef7d4&end=34e6673a0473e90ef01a18eb575392c9e3859747&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=31448badfd74ea72d2c8622cc60d3dca889ef7d4&end=34e6673a0473e90ef01a18eb575392c9e3859747&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=31448badfd74ea72d2c8622cc60d3dca889ef7d4&end=34e6673a0473e90ef01a18eb575392c9e3859747&stat=max-rss)
This benchmark run did not return any relevant results for this metric.

### [Cycles](https://perf.rust-lang.org/compare.html?start=31448badfd74ea72d2c8622cc60d3dca889ef7d4&end=34e6673a0473e90ef01a18eb575392c9e3859747&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
